src/yupSchema.js

```diff
@@ -34,8 +34,45 @@ const validateOnlyStrings = string()
 
 const yupSchemas = {
   text: validateOnlyStrings,
-  select: string().trim().nullable(),
-  radio: string().trim().nullable(),
+  radioOrSelect: (options) =>
+    string()
+      .nullable()
+      .transform((value) => {
+        if (value === '') {
+          return undefined; // [1]
+        }
+        if (options?.includes(null)) {
+          return value;
+        }
+        return value === null
+          ? undefined // [2]
+          : value;
+
+        /*
+          [1] @BUG RMT-518 - explanation from PR#18
+          The "" (empty string) is to keep retrocompatibily with previous version.
+          The "" does NOT match the JSON Schema specs. In the specs the `oneOf` keyword does not allow "" value by default.
+
+          Disallowing "" would be a major BREAKING CHANGE
+          because previously any string was allowed but now only the options[].value are,
+          which means we'd need to also exclude "" from being accepted.   
+          This would be a dangerous change as it can disrupt existing UI Form integrations
+          that might handle empty fields differently ("" vs null vs undefined).
+
+          [2] The null also needs to be always allowed for the same reason.
+              Some consumers (eg Remote) expect empty optional fields to be sent as "null"
+              even when their JSON Schemas are not created correctly (missing an option with const: null).
+              This will allow the JSF to still mark the field as valid (false positive)
+              and let the JSON Schema validator fail.
+
+          We'd need to implement a feature_flag/transition deprecation warning
+          to give devs the time to adapt their integrations before we fix this behavior.
+          Check the PR#18 and tests for more details.
```

Comment:
@gabrielseco @brennj As promised, I've recorded a [public loom](https://www.loom.com/share/3518f2ed732d4699b7ca83f3b220804b) (and added it to the PR description) explaining this workaround with examples.

Hopefully, it will help you better understand/explain this in the future when we come back to this to and create the migration phase to stop accepting `""` and `null` when we shouldn't.